go.mod

```diff
@@ -2,6 +2,8 @@ module github.com/argoproj/argo-cd/v3
 
 go 1.22.0
```

Comment:
Is there any reason to not change the version here as well?

Reply:
To avoid CVE, we can only take care of the Go version used in this repository. The minimum Go version that ArgoCD repository requires is determined by Go toolchain directive. Go directive is a minimum version used by other modules when these try to build within the dependency of ArgoCD. Go1.22 is the version which Go team sill supports since Go1.22 is responsible for compiler [#ref](https://go.dev/doc/install/source#go14).
So I don't change the Go version.

Reply:
I had a [PR](https://github.com/argoproj/argo-cd/pull/21748) open to bump the go version. I haven't had time to investigate the failing tests. I think in general, we should just increase the go version & tools. I don't see the need to maintain different toolchain version.

Unless this provides a way for companies to rebuild the container in a easier way once it is released, I think in general we should aim to try to keep master up to date with the latest versions of everything.

Reply:
Go1.22 is still be supported. If we only support v1.23, the developer that uses CRD of Argo Cd for custom-controller face the build failure. Of course, in many cases, the user can upgrade the Go version themselves, but we can't declare is't 100%.

Reply:
Sorry, I am not following. Are you saying someone using the Argo CD repo as a module and a lower-go version couldn't import it?

Reply:
I agree with @agaudreault, in the worst case, someone would just need to update their go installation to 1.23.X+.

Reply:
@rumstead 
Sorry for confusing you.

> Are you saying someone using the Argo CD repo as a module and a lower-go version couldn't import it?

Yes, What I wanted to say is it.

> I agree with @agaudreault, in the worst case, someone would just need to update their go installation to 1.23.X+.
Okay, well I'll unify go and toolchain directive to 1.23.6